internal/c/libqb.cpp

```diff
@@ -37698,11 +37531,11 @@ void display() {
                         static int32 render_option;
                         static int32 ok;
                         static uint8 *rt_data;
-                        static int32 rt_w, rt_h, rt_pre_x, rt_post_x;
+                        static int32 rt_w, rt_h;
                         render_option = 1;
                         if (rt_data_last)
                             free(rt_data_last);
-                        ok = FontRenderTextUTF32(font[f], &chr_utf32, 1, render_option, &rt_data, &rt_w, &rt_h, &rt_pre_x, &rt_post_x);
+                        ok = FontRenderTextUTF32(font[f], &chr_utf32, 1, render_option, &rt_data, &rt_w, &rt_h);

```

Comment:
I think this is a problem, it happens on the GLUT thread rather than the QB64 thread. Since this will access/modify the `Font` object and `Glyph` cache they need to be thread-safe.

Alternatively we could potentially separate things a bit so that the GLUT thread here gets a _copy_ of the selected `Font`, I'm not really sure which option is easier.

Reply:
Thank you for pointing this out. I'll start woking on adding the thread-safety code in the Glyph caching system. Will update here once I am done.

Reply:
Using mutex in most top-level functions declared in `font.h`.